app/controllers/sentences_controller.php

```diff
@@ -1033,14 +1054,24 @@ public function several_random_sentences()
 
         $randomIds = $this->Sentence->getSeveralRandomIds($lang, $number);
 
-        $this->Session->write('random_lang_selected', $lang);
+        if(is_bool($randomIds) && !$randomIds) {
+            $searchDisabled = !Configure::read('Search.enabled');
+            $this->set('searchProblem', true);
+            if ($searchDisabled) {
```

Comment:
Your if's can be ordered like this:

- if $searchDisable ...
- else if $searchError ...
- else ...

Instead of:

- if $searchError
  - if $searchDisabled ...
  - else ...
- else ...